client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -336,14 +362,14 @@ public void processingFailure(Exception error) {
             state.failConnection(error);
         }
     }
-    
+
     /**
-     * @see SegmentOutputStream#write(java.nio.ByteBuffer,
-     *      java.util.concurrent.CompletableFuture)
+     * @see SegmentOutputStream#write(PendingEvent)
      */
     @Override
     @Synchronized
     public void write(PendingEvent event) throws SegmentSealedException {
+        checkState(!state.isAlreadySealed(), "Segment: {} is already sealed", segmentName);
         ClientConnection connection = getConnection();
```

Comment:
`getConnection()` establishes the connection on receiving "appendSetup" and  re-transmits all the Pending events from the inflight list.  If we add an event to the inflight list before establishing the connection(after a connection drop) we would end up re-transmitting the event. i.e. once as part of AppendSetup callback and then as a separate Append. Hence, i am not planning to swap the two lines.